cmd/forwarder/fixer_test.go

```diff
@@ -39,6 +41,20 @@ func TestFix(t *testing.T) {
 	}
 }
 
+func TestTruncation(t *testing.T) {
+	assert := assert.New(t)
+	message := rfc5424.Message{
+		Priority: rfc5424.Daemon | rfc5424.Info,
+		Timestamp: time.Now(),
+		Hostname: "somehostnamethatsratherlongandneedstobetruncatedsothatitfitsthe48characterparameterasrequestedinthegusitemMikecreated",
+		AppName: "someapp",
+		Message: []byte("Hello, World!"),
+	}
+	message.AddDatum("foo@123", "Revision", "1.2.3.4")
+
+	assert.Equal(len(truncate([]byte(message.Hostname))), 48)
```

Comment:
Don't test the truncate method directly. Instead, invoke `fix`, and verify that what you get back from doing that is exactly what you expect.